utils/ripgrep/wrapper.py

```
@@ -0,0 +1,48 @@
+    raise ValueError("Either a `input.pattern` or `params.pattern` has to be provided")
+
+ignore = snakemake.input.get("ignore")
+if ignore:
+    extra += f" --ignore-file='{ignore}'"
+
+outfile = snakemake.output[0]
+if str(outfile).endswith(".json"):
+    extra += " --json"
+
+compression_fmts = (".gz", ".lzma", ".bz2", ".xz", ".lz4", ".zst")
+if any(str(i).endswith(compression_fmts) for i in snakemake.input):
```

Ruff (0.15.11)

[error] Undefined name `snakemake`

(F821)